symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
```

Comment:
The Intel compiler didn't seem to understand `std::is_constructible`. So again, probably have to use some ifdefs.